cmd/utils/utils.go

```diff
@@ -2,43 +2,58 @@ package utils
 
 import (
 	"errors"
+	"fmt"
 	"log"
 	"os"
+	"path/filepath"
 	"strings"
 	"sync"
 
 	"fyne.io/fyne/v2"
 	"fyne.io/fyne/v2/dialog"
-	"fyne.io/fyne/v2/storage"
 	"fyne.io/fyne/v2/widget"
 )
 
-// Global variables to store the last used directories
-var LastFileDirectory fyne.ListableURI
-var LastFolderDirectory fyne.ListableURI
-
-var Logger *log.Logger
-var LogMessages []string
-var LogMutex sync.Mutex
+// Logger and Log Management
+var (
+	Logger      *log.Logger
+	LogMessages []string
+	LogMutex    sync.Mutex
+)
 
+// InitializeLogger sets up the logger to write to a specified file.
 func InitializeLogger(logFilePath string) error {
+	logDir := filepath.Dir(logFilePath)
+	if _, err := os.Stat(logDir); os.IsNotExist(err) {
+		err = os.MkdirAll(logDir, os.ModePerm)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Overly permissive file permissions allow unauthorized access. Use minimum required permissions: 0400 for read, 0200 for write, 0600 for read/write if only the app user should access the file. Restricting permissions prevents malicious actors from accessing sensitive files.